new_course/1_programming_fundementals/methods.md

```diff
@@ -1,62 +1,268 @@
+
 
 ## Creating a Method
+
 You can create your own custom methods in Ruby like so:
-```(ruby)
+```ruby
 def my_name
   "Joe Smith"
 end
 
 puts my_name #=> "Joe Smith"
 ```
 Let's break it down:
-`def` - is a built in Ruby keyword. It lets Ruby know this is the start of a method *definition*.
+* `def` - is a built in Ruby keyword. It lets Ruby know this is the start of a method *definition*.
 
-`my_name` - is the name of the method. You can name pretty much name it whatever you want. But there are some constraints and conventions which are described in the next section.
+* `my_name` - is the name you are giving to the method. You can pretty much name it whatever you want. But there are some constraints and conventions which are described in the next section.
 
-`"Joe Smith"` - is in the *method body*. The method body is where the logic of your method goes. This particular method will just *return* a string when it's *called*.
+* `"Joe Smith"` - is in the *method body*. This is where all the logic of your method goes. This particular method will just *return* a string when it is *called*.
 
-`end` - as you might have guessed `end` marks the *end* of the method definition. It's another Ruby keyword.
+* `end` - as you might have guessed `end` marks the *end* of the method definition. It's another Ruby keyword.
+
+* To call the method you simply need to use its name, as shown in the last line of the example `puts my_name`
 
-To call the method you simply need to use its name, as shown in the last line of the example `puts my_name`
 
 ## Parameters and Arguments
-* basics about what parameters are
-* default parameters
+
+Of course, not all methods are as simplistic as the example above. After all, what good are methods if you can't interact with them? When you need to return something other than a fixed result, you need methods with parameters. Parameters are variables that your method will receive when it is called. You can have more meaningful and useful interactions with your methods by using parameters to make them more versatile.
+
+```ruby
+def greet(name)
+  "Hello " + name + "!"
+end
+
+puts greet("John") #=> Hello John!
+```
+
+In this simple example, `name` is a parameter that the `greet` method uses to return a more specific greeting. The method is called with the argument `"John"`, and returns the string "Hello John!"
+
+You might be confused when to use the term *argument* vs. *parameter*. Don't be: Parameters effectively act as Placeholder variables in the template that is your method, while Arguments are the Actual variables that get passed to the method when it is called. The two terms are commonly used interchangeably, so don't worry too much about it.
```

Comment:
There is some unnecessary capitalisation of words here such as placeholder and actual.